### PR TITLE
Update the site for the 0.9.0 release

### PR DESCRIPTION
**Pull request by rdblue**
This updates the site for the 0.9.0 release:
* Add 0.9.0 to the releases page
* Point Javadoc redirect to 0.9.0
* Add warnings to Spark documentation
* Update the Getting Started page to use Spark 3 SQL instead of the Java API and DataFrames
* Add Spark catalogs to Configuration
* Remove duplicate Spark API Quickstart with the same content as Java API Quickstart
* Minor updates to CSS to reduce density

### CODE REVIEW

File: site/docs/spark.md [hunk 1]
```diff
@@ -286,6 +286,11 @@ val df = spark.read
     .table("prod.db.table")
 ```
 
+!!! Warning
+    When reading with DataFrames in Spark 3, use `table` to load a table by name from a catalog.
+    Using `format("iceberg")` loads an isolated table reference that is not refreshed when other queries update the table.
```
Comment (rdblue) [inline site/docs/spark.md:291]: @dongjoon-hyun, this is the warning I've added to make people aware of the issues with the `DataFrameReader`. Please take a look if you have time. There is also one below for the v1 `DataFrameWriter` API.

Comment (dongjoon-hyun) [reply site/docs/spark.md:291]: Thank you, @rdblue . It looks good to me.

File: site/docs/spark.md [hunk 2]
```diff
@@ -353,6 +358,13 @@ To replace data in the table with the result of a query, use `INSERT OVERWRITE`.
 
 The partitions that will be replaced by `INSERT OVERWRITE` depends on Spark's partition overwrite mode and the partitioning of a table.
 
+!!! Warning
+    Spark 3.0.0 has a correctness bug that affects dynamic `INSERT OVERWRITE` with hidden partitioning, [SPARK-32168][spark-32168].
+    For tables with [hidden partitions](../partitioning), wait for Spark 3.0.1.
```
Comment (dongjoon-hyun) [inline site/docs/spark.md:363]: Ya. We should release Apache Spark 3.0.1 soon for the users.

File: site/docs/spark.md [hunk 3]
```diff
@@ -432,6 +444,13 @@ Spark 3 introduced the new `DataFrameWriterV2` API for writing to tables using d
     - `df.writeTo(t).append()` is equivalent to `INSERT INTO`
     - `df.writeTo(t).overwritePartitions()` is equivalent to dynamic `INSERT OVERWRITE`
 
+The v1 DataFrame `write` API is still supported, but is not recommended.
+
+!!! Warning
+    When writing with the v1 DataFrame API in Spark 3, use `saveAsTable` or `insertInto` to load tables with a catalog.
+    Using `format("iceberg")` loads an isolated table reference that will not automatically refresh tables used by queries.
```
Comment (dongjoon-hyun) [inline site/docs/spark.md:451]: Thanks, @rdblue !

**Comment by rdblue** [comment]
Thanks for reviewing, @dongjoon-hyun and @rdsr!

**Comment by rdblue** [comment]
Merging since this doesn't affect the build and CI was flaky.

**Pull request merged**